### PR TITLE
Fix #104 - add support for 3.12

### LINKED ISSUE #104: Add support for jooq 3.12.x
Hello,

I'm using your plugin wiht jooq 3.12.x and I get a lot of warnings of deprecated fields.
My build is failing right not but I am not sure if it is because of the plugin:

```
ls.JooqLogger warn
WARNING: DEPRECATED               : The <expression/> element in <forcedType/> is deprecated. Use <includeExpression/> instead: <userType>java.util.Map</userType><binding>gr8pi.jooq.databindings.jackson.converters.AnswersBinding</binding><includeExpression>.*\.answers.*</includeExpression><types>.*jsonb.*</types><nullability>ALL</nullability><objectType>ALL</objectType>

```

### PR DESCRIPTION
**Pull request by ieugen**
./gradlew clean check testAll ran successfully

### CODE REVIEW

**Comment by dgliosca** [comment]
Hi there, have you got any timeline on when this is going to be mainstream?

**Comment by alexanderjohn** [comment]
@etiennestuder any progress on this?  same question goes with the other open PR's and issues.  the project looks to be losing a bit of its steam.

File: src/test/groovy/nu/studer/gradle/jooq/JooqFuncTest.groovy [hunk 1]
```diff
@@ -62,7 +62,8 @@ class JooqFuncTest extends BaseFuncTest {
         def result = runWithArguments('build')
 
         then:
-        result.task(':generateSampleJooqSchemaSource').outcome == TaskOutcome.SUCCESS
+        def outcome = result.task(':generateSampleJooqSchemaSource').outcome
+        (outcome == TaskOutcome.SUCCESS) || (outcome == TaskOutcome.UP_TO_DATE)
```
Comment (etiennestuder) [inline src/test/groovy/nu/studer/gradle/jooq/JooqFuncTest.groovy]: That's hiding the underlying problem instead of fixing it.

**Comment by etiennestuder** [comment]
Updated in 4.0

**Pull request closed without merging**